p2p/swarm.go

```diff
@@ -284,6 +284,32 @@ func (s *swarm) SendMessage(peerPubkey p2pcrypto.PublicKey, protocol string, pay
 	return s.sendMessageImpl(peerPubkey, protocol, service.DataBytes{Payload: payload})
 }
 
+func (s *swarm) RetrySend(peerPubKey p2pcrypto.PublicKey, message []byte) error {
```

Comment:
this shouldn't be an exported method

Reply:
also please add tests for your change

Reply:
> this shouldn't be an exported method
Since `RetrySend` will need to establish a connection and session using swarm, shouldn't it be a exported method?

Reply:
nope, the only consideration about exporting or not exporting is whether we want any client to use this function, which we don't. this is why this issue is called *abstract* message retry.

Reply:
Okay. I have worked on it and is now no longer an exported method. Just need to write the tests now.